ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py

```diff
@@ -351,15 +351,18 @@ def _switch_data_node_group_nodes(self):
         return get(self._switch_data, "node_group.nodes", default=[])
 
     @cached_property
-    def group(self):
+    def group(self) -> str:
         """
         switch.group set to "node_group" name or None
         """
         return get(self._switch_data, "group")
 
     @cached_property
-    def id(self):
-        return get(self._switch_data_combined, "id", required=True)
+    def id(self) -> int | None:
+        """
+        Make id not required for switches that would not use it
```

Comment:
this docstring will not make much sense after this PR.

Reply:
good point!